ios/sound.mm

```diff
@@ -24,209 +24,277 @@
 #include "sound.h"
 #include <AVFoundation/AVFoundation.h>
 
-
-/* Implementation *************************************************************/
-CSound::CSound ( void           (*fpNewProcessCallback) ( CVector<short>& psData, void* arg ),
-                 void*          arg,
-                 const QString& strMIDISetup,
-                 const bool     ,
-                 const QString& ) :
-    CSoundBase ( "CoreAudio iOS", fpNewProcessCallback, arg, strMIDISetup ),
-    midiInPortRef ( static_cast<MIDIPortRef> ( NULL ) )
+#define kOutputBus 0
+#define kInputBus  1
+
+/**
+ This callback is called when sound card needs output data to play. And because Jamulus use the same buffer to store input and output data (input is
+ sent to server, then output is fetched from server), we actually use the output callback to read inputdata first, process it, and then copy the
+ output fetched from server to ioData, which will then be played.
+ */
+static OSStatus recordingCallback ( void*                       inRefCon,
```

Comment:
Why isn't this a method on the application object?

Reply:
I was following an iOS sound code example, didn't really think about it. Thanks to you, I put the function inside CSound, and it still works!

Reply:
The code here on github still has the method as a static global.  Did you miss your change getting committed and pushed?

Reply:
I didn't commit, working on --amend